spec/unit/shoulda/matchers/active_model/validate_length_of_matcher_spec.rb

```diff
@@ -370,6 +370,190 @@ def configure_validation_matcher(matcher)
+          not_to validate_length_of(:children).is_at_most(3).with_long_message(/.*/)
+      end
+
+      it 'rejects ensuring a higher maximum length with any message' do
+        expect(validating_association_length(maximum: 4)).
+          not_to validate_length_of(:children).is_at_most(5).with_long_message(/.*/)
+      end
+
+      it 'does not override the default message with a blank' do
+        expect(validating_association_length(maximum: 4)).
+          to validate_length_of(:children).is_at_most(4).with_long_message(nil)
+      end
+    end
+
+    context 'an attribute with a required exact length' do
+      it 'accepts ensuring the correct length' do
+        expect(validating_association_length(is: 4)).
+          to validate_length_of(:children).is_equal_to(4)
+      end
+
+      it 'rejects ensuring a lower maximum length with any message' do
+        expect(validating_association_length(is: 4)).
+          not_to validate_length_of(:children).is_equal_to(3).with_message(/.*/)
+      end
+
+      it 'rejects ensuring a higher maximum length with any message' do
+        expect(validating_association_length(is: 4)).
+          not_to validate_length_of(:children).is_equal_to(5).with_message(/.*/)
+      end
+
+      it 'does not override the default message with a blank' do
+        expect(validating_association_length(is: 4)).
+          to validate_length_of(:children).is_equal_to(4).with_message(nil)
+      end
+    end
+  end
+
+  context 'when validating has many through associations' do
+    context 'an association with a non-zero minimum length validation' do
+      it 'accepts ensuring the correct minimum length' do
+        expect(validating_through_association_length(minimum: 4)).
+          to validate_length_of(:children).is_at_least(4)
+      end
+
+      it 'rejects ensuring a lower minimum length with any message' do
+        expect(validating_through_association_length(minimum: 4)).
+          not_to validate_length_of(:children).is_at_least(3).with_short_message(/.*/)
+      end
+
+      it 'rejects ensuring a higher minimum length with any message' do
+        expect(validating_through_association_length(minimum: 4)).
+          not_to validate_length_of(:children).is_at_least(5).with_short_message(/.*/)
+      end
+
+      it 'does not override the default message with a blank' do
+        expect(validating_through_association_length(minimum: 4)).
+          to validate_length_of(:children).is_at_least(4).with_short_message(nil)
+      end
+
+      it 'fails when used in the negative' do
+        assertion = lambda do
+          expect(validating_through_association_length(minimum: 4)).
+            not_to validate_length_of(:children).is_at_least(4)
+        end
+
+        message = <<-MESSAGE
+Expected Parent not to validate that the length of :children is at least
+4, but this could not be proved.
+  After setting :children to ‹[#<Child id: nil>, #<Child id: nil>,
```

Comment:
This is out of scope for this PR, but this error message seems strange to me. If a model doesn't validate that an association matches a certain length, then it should allow a value that would have been invalid with the validation in place. I don't see why having four children should be invalid in this case. If anything having _three_ children should be _valid_. Of course that counts as breaking behavior, and it also could be inconsistent with how "negative" versions of other matchers work. It would have to be changed in concert with other matchers, probably. So I'm not eager to change it now, but I thought I'd point it out.

Reply:
I share your sentiment. The "negative" message sounded very strange to me when I first started working on shoulda-matchers. However, I attributed this perception to my non-native English background. Reading it in Portuguese, my primary language, sounded weird. Should we open an issue to discuss this?

Reply:
Yeah, that sounds like a good idea!